build_vanilla_app.py

```diff
@@ -0,0 +1,152 @@
+
+    # Hidden options
+    parser.add_argument('--step', default='launch', choices=['launch', 'publish'], help=argparse.SUPPRESS)
+    parser.add_argument('--apks-dir', help=argparse.SUPPRESS)
+    parser.add_argument('--google-api-key', help=argparse.SUPPRESS)
+
+    options = parser.parse_args()
+    options.package_name = "org.kiwix.kiwixmobile"
+
+    options.base_version = "{}{}".format(
+        datetime.date.today().strftime('%y%j'),
+        options.extra_code)
+
+    return options
+
+def do_publish(options):
+    googleService = Google(options)
+    with googleService.new_request():
+        versionCodes = []
+        for apk in glob.iglob(options.apks_dir+"/*.apk"):
+            result = googleService.upload_apk(apk)
+            versionCodes.append(result['versionCode'])
+        googleService.publish_release(options, versionCodes)
+
+
+class Google:
+    def __init__(self, options):
+        scope = 'https://www.googleapis.com/auth/androidpublisher'
+        key = options.google_api_key
+        credentials = ServiceAccountCredentials.from_json_keyfile_name(
+            key,
+            scopes=[scope])
+
+        http = httplib2.Http()
+        http = credentials.authorize(http)
+
+        self.service = build('androidpublisher', 'v2', http=http)
+        self.packageName = options.package_name
+        self.edit_id = None
+
+    @contextmanager
+    def new_request(self):
+        edit_request = self.service.edits().insert(
+            packageName=self.packageName,
+            body={})
+        result = edit_request.execute()
+        print("create", result)
+        self.edit_id = result['id']
+
+        yield
+
+        commit_request = self.service.edits().commit(
+            editId=self.edit_id,
+            packageName=self.packageName)
+        result = commit_request.execute()
+        print("commit", result)
+
+        self.edit_id = None
+
+    def make_request(self, section, method, **kwargs):
+        request_content = self._build_request_content(kwargs)
+        _section = getattr(self._edits, section)()
+        _method = getattr(_section, method)
+        print(">", request_content)
+        request = _method(**request_content)
+        result = request.execute()
+        print("<", result)
+        return result
+
+    def _build_request_content(self, kwargs):
+        d = kwargs.copy()
+        d['editId'] = self.edit_id
+        d['packageName'] = self.packageName
+        return d
+
+    @property
+    def _edits(self):
+        return self.service.edits()
+
+    def upload_apk(self, apk_file):
+        return self.make_request('apks', 'upload',
+            media_body=apk_file)
+
+    def publish_release(self, options, versionCodes):
+        return self.make_request('tracks', 'update',
+            track="internal",
+            body={'versionCodes': versionCodes})
+
+
+def gen_version_code(platform_index, base_version):
```

Comment:
Why duplicate all those functionalities (`gen_version_code`, `Google`, `do_publish`, ...) ?
They exist already in `build_custom_app.py`, you should reuse them. (and if needed, move them in some kind of utility python module)